ruby/ql/lib/codeql/ruby/ast/Control.qll

```diff
@@ -371,12 +396,37 @@ class CaseExpr extends ControlExpr, TCaseExpr {
     or
     pred = "getBranch" and result = this.getBranch(_)
     or
-    pred = "getWhenBranch" and result = this.getWhenBranch(_)
-    or
     pred = "getElseBranch" and result = this.getElseBranch()
   }
 }
 
+private class CaseWhenExpr extends CaseExpr, TCaseExpr {
```

Comment:
For other classes we have often used this pattern:

`internal/Control.qll`:
```ql
abstract class CaseExprImpl extends ControlExpr, TCase {
  abstract Expr getValue();

  abstract Expr getBranch(int n);
}

class CaseWhenExpr extends CaseExprImpl, TCaseExpr {
  ...
}

class CaseMatch extends CaseExprImpl, TCaseMatch {
  ...
}
```

`Control.qll`:
```ql
class CaseExpr extends ControlExpr instanceof CaseExprImpl {
  final Expr getValue() { result = super.getValue() }

  final Expr getBranch(int n) { result = super.getBranch(n) }

  ...
}
```

This has the advantage that we can refer to `CaseWhenExpr` and `CaseMatch` in internal code (e.g. `ControlFlowGraphImpl.qll`), and it prevents overriding of the two predicates in the external layer. If you make this change, remember to (privately) import `internal/Control.qll` into `Control.qll`.